src/coreclr/vm/prestub.cpp

```diff
@@ -2562,6 +2562,10 @@ static PCODE PreStubWorker_Preemptive(
         // more we can do except fail fast. The reverse P/Invoke isn't
         // going to work.
         CREATETHREAD_IF_NULL_FAILFAST(currentThread, W("Failed to setup new thread during reverse P/Invoke"));
+
+        // [TODO] Ensure thread is in Preemptive mode.
```

Comment:
I assume this is a hack that will be reverted.

Reply:
This isn't a hack, at least I don't think it is. There is something wrong in this code path and I'm not sure yet what it is, but it seems like a subtle bug. We are creating a thread above in `CREATETHREAD_IF_NULL_FAILFAST` and it is in COOP mode. I'm unclear how anything I am doing can influence this at present. I put this in a separate commit to investigate tomorrow.

Reply:
I think you are missing `HasStarted`  call when setting up the new thread. Creating a new thread in native CoreCLR is a complicated handshake. You may want to create the thread on the managed side - it is much easier and less bug prone to do that way, and it can be reused for NAOT.

Reply:
This worked. Thanks.